### ISSUE TITLE
Getting HTTP 413 Errors for Image upload when files are smaller than the max setting.

### ISSUE THREAD

**Issue by u2g-tg** [bitnami/bitnami-docker-discourse#120]
We have a working deployment of Discourse, and have changed the Max Image Upload Setting to `4096`. We are using the s3 upload plugin, and it has also been proven working (with small files)

The help text reads "The maximum attachment files upload size in kB. This must be configured in nginx (client_max_body_size) / apache or proxy as well."

However when you try to add an image larger than 1MB in a post, you get a popup saying that its larger than the designated size. In addition the browser network debug has a post to 'uploads.json' which fails with a 413 HTTP error code. 

Research tells me that Bitnami Discourse is deployed with Nginx/Phusion Passenger Standalone, and I cannot find how to increase the max client body to prove that it is/isn't Nginx stopping the upload. The production.log doesn't help either. 

Please help. How can I increase the upload limit above 1MB in the web server.

**Comment by andresbono** [comment]
If you get a popup message in the web application, it is likely that the misconfiguration is in Discourse but not in Nginx.

Did you start the deployment from scratch when you changed the max image upload setting?

**Comment by u2g-tg** [comment]
The discourse process had been restarted multiple times with the 4096kb upload setting, but consistently showed the popup when the file to upload was larger than 1024kb. 

This is the popup being triggered in the UI https://github.com/discourse/discourse/blob/f1cd0d63883d9c0e5becf780c82a749a9da0e23b/app/assets/javascripts/discourse/lib/uploads.js.es6#L263

I find it suspicious that the Nginx default client_max_body_size is 1Mb 
http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size

But I know that this Bitnami image is using Phusion Passenger Standalone, and having grepped the docker image files, the only nginx config file I can find from Phusion, puts the client_max_body_size at 1024MB.

**Comment by javsalgar** [comment]
Hi,

I was unable to reproduce the issue. I just uploaded a 4MB image without problems

![image](https://user-images.githubusercontent.com/1133777/69956817-5c8b3900-1501-11ea-9008-8f220d76c24e.png)


Could you confirm you are using the latest version of the container?

**Comment by u2g-tg** [comment]
We are currently using this Container as a base, with the discourse-solved plugin unzipped into it

https://hub.docker.com/layers/bitnami/discourse/2.3.6-debian-9-r19/images/sha256-547cd9f7f7fa9b849e27b94da4aa8bc1a7487714499aaed8f7bc4e0959fce77f

I ought to test whether we get the same behaviour with the s3 image storage plugin disabled.

**Comment by javsalgar** [comment]
Please let us know if that's the case

**Comment by u2g-tg** [comment]
OK, I figured it out, the Bitnami Discourse image is fine (as expected when the Phusion Passenger sets the client_max_body_size to '1024m'.

We are running in Kubernetes with Nginx Ingress, and the ingress reverse proxy has a default 1MB client_max_body_size. 

FYI, fixed it with an Annotation on our forum ingress definition: 
nginx.ingress.kubernetes.io/proxy-body-size: 12M 

(Nginx-Ingress uses a slightly different key for the same result) 

Closing issue.
